src/mumble/PipeWire.h

```diff
@@ -0,0 +1,127 @@
+// Copyright 2021 The Mumble Developers. All rights reserved.
+// Use of this source code is governed by a BSD-style license
+// that can be found in the LICENSE file at the root of the
+// Mumble source tree or at <https://www.mumble.info/LICENSE>.
+
+#ifndef MUMBLE_MUMBLE_PIPEWIRE_H
+#define MUMBLE_MUMBLE_PIPEWIRE_H
+
+#include "AudioInput.h"
+#include "AudioOutput.h"
+
+#include <QWaitCondition>
+
+class PipeWireInit;
+
+struct pw_buffer;
+struct pw_loop;
+struct pw_properties;
+struct pw_stream;
+struct pw_stream_events;
+struct pw_thread_loop;
+
+struct spa_dict;
+struct spa_pod;
+
+class PipeWireEngine {
+private:
+	Q_DISABLE_COPY(PipeWireEngine)
+protected:
+	bool m_ok;
+	pw_loop *m_loop;
+	pw_stream *m_stream;
+	pw_thread_loop *m_thread;
+	std::unique_ptr< pw_stream_events > m_events;
```

Comment:
Is `pw_stream_events` an array? If so we can't use `unique_ptr` like this as we'd have to use the array specialization of the template, which I don't think will be selected when using a forward declared type :thinking:

Reply:
It's a structure.